### ISSUE TITLE
Drawer takes up the entire width of the screen (480x800, portrait mode)

### ISSUE THREAD

**Issue by stepanov-standy** [mikepenz/MaterialDrawer#319]
Hi Mike!

First of all I would like to thank you. You've implemented awesome library!

I've successfully integrated it in my application and it works perfectly. During my tests on different devices I've found only one thing that doesn't work out of the box. I am talking about drawer that takes up the entire width of the screen when I test application on emulator:
Android Version: 2.3.7 (API 10)
Device: Google Nexus One
Screen resolution: 480x800
Mode: portrait

In other words the drawer takes up all 480 pixels of the screen instead of having (according to Google [Guideline] (http://www.google.com/design/spec/patterns/navigation-drawer.html#navigation-drawer-specs)) "side nav equals the screen width minus the height of the action bar. 56dp from the right edge of the screen."

Am I right that developer should worry about this by himself? Do you have any build-in logic related to this?

**Comment by mikepenz** [comment]
@stepanov-standy seems an issue for this device. there is a logic inside the drawer calculating the correct width for all devices (tablest, landscape, etc...) i will take a look at it, and why it is wrong for a Nexus One.

**Comment by mikepenz** [comment]
Just in case this is the code:

```java
public static int getOptimalDrawerWidth(Context context) {
        int possibleMinDrawerWidth = UIUtils.getScreenWidth(context) - UIUtils.getThemeAttributeDimensionSize(context, android.R.attr.actionBarSize);
        int maxDrawerWidth = context.getResources().getDimensionPixelSize(R.dimen.material_drawer_width);
        return Math.min(possibleMinDrawerWidth, maxDrawerWidth);
    }
```

I assume on API Level 10 the `actionBarSize` resource return `0`

**Comment by mikepenz** [comment]
@stepanov-standy will be fixed in the next release. As i mentioned before the issue is that on API Level 10 it was not able to calculate the ActionBarSize

**Comment by stepanov-standy** [comment]
Thank you Mike for your quick response )
